src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -79,6 +79,8 @@ public sealed class BinaryLogger : ILogger
         //    - new record kinds: BuildCheckMessageEvent, BuildCheckWarningEvent, BuildCheckErrorEvent,
         //    BuildCheckTracingEvent, BuildCheckAcquisitionEvent, BuildSubmissionStartedEvent
 
+        // MAKE SURE YOU KEEP BuildEventArgsWriter AND StructuredLogViewer.BuildEventArgsWriter IN SYNC WITH THE CHANGES ABOVE.
```

Comment:
I think giving a brief explanation on the why we need to do that would be nice here.